app/scripts/directives/editEnvironmentFrom.js

```diff
@@ -134,7 +134,11 @@
       return referenceValue;
     };
 
-    ctrl.checkEntries = function(option) {
+    ctrl.checkEntries = function(option, entrySelectedEnvFrom) {
+      if(_.isEqual(option, entrySelectedEnvFrom)) {
```

Comment:
Does `option === entrySelectedEnvFrom` work or is a copy?

Reply:
So it might... but the data within "entrySelectedEnvFrom" is "moved" over so kind of a copy ... I had thought about placing a check for ```option.kind === entrySelectedEnvFrom.kind``` to avoid firing the ```_.isEqual``` every time

Reply:
You could check the UIDs, something like (untested)

```js
if (entrySelectedEnvFrom.metadata.uid === option.metadata.uid) {
  return false;
}
```

assuming the neither is ever null.

Reply:
Secrets and config maps can contains megabytes of data, so would be good to avoid a deep comparison if possible.

Reply:
That works, and understood on the object complexity... Went ahead and checked the ```===``` comparison, it does appear to function as expected, they evaluate as equal. Was a little unsure since one of the parameters is being pulled through a nested ```ng-repeat```